Add Core Notes July 26, 2024 (#4120)
tock/tock: doc/wg/core/notes/core-notes-2024-07-26.md

* Leon: That was opened by a labmate of ours, it stems from a project on timing isolation. I will be managing this for the most part. Implementation is driving forward regardless because this is a research project, and also outlines how invasive changes would be. In practice we won't move forward with this before a descriptive document is ready.
* Hudson: https://github.com/tock/tock/pull/4075
* Leon: Should we merge it on the call? It is last call
* Everyone: Yes!
* Brad: as an aside, it would be nice to have a Tock community bot. For example I opened this PR, but then could not approve it even though other people contributed many of the commits.
* Alyssa: Could we allow people with write access to the Tock repo directly merge without approval? For community PR cases like this?
* Brad: I think that would be fine, but the issue I am talking about is sort of the opposite – for example I could not put a "changes requested" review on my own PR if Amit adds commits and then approves. Probably not a thing we need to discuss, just a workflow mismatch.

## Dynamic allocation in libtock-rs

> [meta: I was notetaker for this meeting so this initial part is copied from my

Reviewer: ```suggestion
[meta: Hudson was notetaker for this meeting so this initial part is copied from Hudson's
```

# Tock Meeting Notes 2024-07-26

## Attendees
- Hudson Ayers
- Pat Pannuto
- Brad Campbell
- Leon Schuermann
- Benjamin Prevor
- Tyler Potyondy
- Alyssa Haroldsen
- Andrew Imwalle
- Alexandru Radovici

## Updates

* Ben: We have a new employee of the Tock foundation – Benjamin Prevor, he is
  working on the treadmill testing infrastructure. Before this he was a full
  time Haskell engineer.
* Pat: We have a libtock-c PR pulling in a Lora example that brings in GPL
  code. It is constrained to an example so hopefully contained and does not
  cause issues, but wanted to flag it so people could check it out.
  https://github.com/tock/libtock-c/pull/456

## Outstanding PR triage

> *Hudson: https://github.com/tock/tock/pull/4109 *Brad: Lets mark that as

Reviewer: This doesn't format properly as bullet points.

philosophy etc. before diving into it. *Leon: That was opened by a labmate of
ours, it stems from a project on timing isolation. I will be managing this for
the most part. Implementation is driving forward regardless because this is a
research project, and also outlines how invasive changes would be. In practice
we won't move forward with this before a descriptive document is ready.
*Hudson: https://github.com/tock/tock/pull/4075 *Leon: Should we merge it on
the call? It is last call *Everyone: Yes! *Brad: as an aside, it would be nice
to have a Tock community bot. For example I opened this PR, but then could not
approve it even though other people contributed many of the commits. *Alyssa:
Could we allow people with write access to the Tock repo directly merge without
approval? For community PR cases like this? *Brad: I think that would be fine,
but the issue I am talking about is sort of the opposite – for example I could
not put a "changes requested" review on my own PR if Amit adds commits and then
approves. Probably not a thing we need to discuss, just a workflow mismatch.

## Dynamic allocation in libtock-rs

> [meta: I was notetaker for this meeting so this initial part is copied from my

Reviewer 2: It's not clear outside the PR who "I" is, so just replace with Hudson maybe?
